src/test.ts

```diff
@@ -104,9 +109,39 @@ export function runTest(testDirectory: string, rulesDirectory?: string | string[
             };
         });
 
+        // test against fixed files
+        let fixedFileText: string;
+        let newFileText: string;
+        try {
+            const fixedFile = fileToLint.replace(/\.lint$/, FIXES_FILE_EXTENSION);
+            const stat = fs.statSync(fixedFile);
+            if (stat.isFile()) {
+                fixedFileText = fs.readFileSync(fixedFile, "utf8");
+                // accumulate replacements
+                const replacements = failures.reduce((acc, failure) => {
+                    const fixes = failure.getFixes();
+                    if (fixes.length > 0) {
+                        return acc.concat(fixes[0].replacements);
+                    } else {
+                        return acc;
+                    }
+                }, [] as IReplacement[]);
+                // sort in reverse so that diffs are properly applied
+                replacements.sort((a, b) => (b.start + b.length) - (a.start + a.length));
+                newFileText = replacements.reduce((acc, r) => {
```

Comment:
at some point, other places (eg. the ProseFormatter) will want to be able to apply fixes, so this probably belongs somewhere else. It could be refactored to another location in a later PR, when there is a second caller who wants to share it.